### PR TITLE
Add io.github.libredirect.frontends-manager

### PR DESCRIPTION
**Pull request by ManeraKai**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

I need to access `docker` so I used the `org.freedesktop.Flatpak` D-bus interface.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.libredirect.frontends-manager.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+id: org.libredirect.frontends-manager
+runtime: org.gnome.Platform
+runtime-version: '42'
```
Comment (hfiguiere) [inline org.libredirect.frontends-manager.yml:3]: this runtime has been EOL for a year.

File: org.libredirect.frontends-manager.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+  - org.freedesktop.Sdk.Extension.rust-nightly
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=xdg-data
```
Comment (hfiguiere) [inline org.libredirect.frontends-manager.yml:17]: why do you need access to this. It's not allowed by default and usually unecessary.

File: org.libredirect.frontends-manager.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  - --device=dri
+  - --talk-name=org.freedesktop.Flatpak
```
Comment (hfiguiere) [inline org.libredirect.frontends-manager.yml:19]: you say you use it for docker, but what if docker doesn't exist.

Why can't you instead bundle the docker command that connect to the docker socket?

File: org.libredirect.frontends-manager.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - type: git
+        url: https://github.com/libredirect/frontends_manager.git
+        branch: main
+        tag: v1.0.0
+        commit: 1ab80c9068698db809151392ef2c9178d95d7dc7
+        x-checker-data:
+          type: git
+          tag-pattern: ^v((?:\d+.)*\d+)$
+      - type: file
+        path: org.libredirect.frontends-manager.desktop
```
Comment (hfiguiere) [inline org.libredirect.frontends-manager.yml:34]: you build system build a debian package without a .desktop file?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
there is no appstream file installed.

File: org.libredirect.frontends-manager.yml (hunk 3 above)
Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:19]: Well that would be better. I'll try to implement that.

File: org.libredirect.frontends-manager.yml (hunk 4 above)
Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:34]: I'm just using the debian build bundle bc it's the most straight forward way to compile a binary from tauri.

File: org.libredirect.frontends-manager.yml (hunk 1 above)
Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:3]: Bc tauri still doesn't support newer webgtk.

File: org.libredirect.frontends-manager.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  - --device=dri
+  - --talk-name=org.freedesktop.Flatpak
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+  - name: libredirect-frontends-manager
+    sources:
+      - type: git
+        url: https://github.com/libredirect/frontends_manager.git
+        branch: main
```
Comment (hfiguiere) [inline org.libredirect.frontends-manager.yml:27]:
```suggestion
```

NEVER a branch.

File: org.libredirect.frontends-manager.yml (hunk 4 above)
Comment (hfiguiere) [reply org.libredirect.frontends-manager.yml:34]: I would expect that this have a .desktop file. After all this is a 20+ year old standard and shipping an app without one is a bug. And if there is one why isn't it used.

File: org.libredirect.frontends-manager.yml (hunk 3 above)
Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:19]: Apparently with Docker Desktop there's no docket socket. Even [Whaler](https://flathub.org/apps/com.github.sdv43.whaler) isn't working. I use docker to download images and run/stop them.

Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:19]: I also have a way to detect if docker doesn't exist and notify the user.

File: org.libredirect.frontends-manager.yml (hunk 1 above)
Comment (bbhtt) [reply org.libredirect.frontends-manager.yml:3]: > Bc tauri still doesn't support newer webgtk.

Are you able to use 2.0 alpha builds in the project?

Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml:3]: I'll try to upgrade the gnome platform version as soon as 2.0 gets stable.

File: org.libredirect.frontends-manager.yml (hunk 4 above)
Comment (ManeraKai) [reply org.libredirect.frontends-manager.yml]: Alright removed the desktop file from this repo.

**Comment by ManeraKai** [comment]
> there is no appstream file installed.

Added it.

File: org.libredirect.frontends_manager.metainfo.xml
Comment (bbhtt) [inline org.libredirect.frontends_manager.metainfo.xml:1]:
```
$ appstream-util validate org.libredirect.frontends_manager.metainfo.xml        

org.libredirect.frontends_manager.metainfo.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• file-invalid          : <screenshot> failed to load [https://raw.githubusercontent.com/flathub/org.libredirect.frontends_manager/screenshots/Main%20Window.png]
• tag-missing           : <release> required
• tag-missing           : <url> is not present
Validation of files failed
```

You need to fix these.

File: org.libredirect.frontends_manager.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  
+  <name>Libredirect Frontends Manager</name>
+  <summary>Run privacy frontends on your computer</summary>
+  
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-3.0-or-later</project_license>
+  
+  <description>
+    <p>
+      Run privacy-friendly frontends locally on your computer rather than relying on 3rd party instances. A much better trust-less model. For anonymizing your IP, use Lokinet or a VPN.
```
Comment (bbhtt) [inline org.libredirect.frontends_manager.metainfo.xml:13]: Nitpick: Please try to follow the quality guidelines eg.

https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/quality-guidelines#no-weird-formatting-1

> summary [...] shouldn't end with a full stop

Not a technical requirement but improves presentation.

File: org.libredirect.frontends_manager.metainfo.xml
Comment (bbhtt) [reply org.libredirect.frontends_manager.metainfo.xml:1]: Appdata should ideally belong in the upstream project and handled by build-system

Comment (ManeraKai) [reply org.libredirect.frontends_manager.metainfo.xml:1]: Fixed the issues.

File: org.libredirect.frontends_manager.metainfo.xml (hunk 6 above)
Comment (ManeraKai) [reply org.libredirect.frontends_manager.metainfo.xml]: Formatted it better.

File: org.libredirect.frontends_manager.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      env:
+        npm_config_nodedir: /usr/lib/sdk/node16
+        CARGO_HOME: /run/build/libredirect-frontends-manager/cargo
+        npm_config_offline: 'true'
+        npm_config_cache: /run/build/libredirect-frontends-manager/flatpak-node/npm-cache
+    build-commands:
+      - npm ci --offline
+      - cargo --offline fetch --manifest-path src-tauri/Cargo.toml
+      - npm run tauri build -- -d -b deb
+      - install -Dm644 -t /app/share/applications/ src-tauri/target/debug/bundle/deb/*/data/usr/share/applications/*
```
Comment (hfiguiere) [inline org.libredirect.frontends_manager.yml:46]: why `*` and not just `*.desktop`. Because that's only for desktop files.

File: org.libredirect.frontends_manager.metainfo.xml
Comment (hfiguiere) [inline org.libredirect.frontends_manager.metainfo.xml:1]: this file is never installed.

File: org.libredirect.frontends_manager.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.libredirect.frontends_manager</id>
+
+  <name>Libredirect Frontends Manager</name>
+  <summary>Run privacy-friendly frontends on your computer</summary>
```
Comment (hfiguiere) [inline org.libredirect.frontends_manager.metainfo.xml:6]: I love "privacy-friendly" but then the first thing done is punching a hole in the sandbox to allow remote executiion.

File: org.libredirect.frontends-manager.yml (hunk 2 above)
Comment (hfiguiere) [reply org.libredirect.frontends-manager.yml]: I have asked again about this.

File: org.libredirect.frontends-manager.yml (hunk 3 above)
Comment (hfiguiere) [reply org.libredirect.frontends-manager.yml]: but it this package require something external like docker, then the point is moot.

also why not podman or more opensource friendle solution

Comment (hfiguiere) [reply org.libredirect.frontends-manager.yml]: > Apparently with Docker Desktop there's no docket socket

it needs to be exposed inside the flatpak. Which is also terrible, but still better.

File: org.libredirect.frontends_manager.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+  - org.freedesktop.Sdk.Extension.rust-nightly
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=xdg-data
```
[the remark this answers is not in the document]
Comment (ManeraKai) [reply org.libredirect.frontends_manager.yml:18]: Sorry, I xdg-data to download the frontends and store config files.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.libredirect.frontends_manager.yml:18]: I still don't understand why you need it.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.libredirect.frontends_manager.yml:18]: you likely don't. otherwise you likely need to fix it.

[the remark this answers is not in the document]
Comment (bbhtt) [reply org.libredirect.frontends_manager.yml:18]: > Sorry, I need xdg-data to download the frontends and store config files.

Why can't you store them in `/home/wirt/.var/app/<app-id>/data`? That's what `$XDG_DATA_HOME` is set in the sandbox. What you're doing right now grants access to the `$XDG_DATA_HOME` on host ie. `~/.local/share` which is a filesystem escape.

You're app should support the XDG basedir specification to be well behaved in this regard.

[the remark this answers is not in the document]
Comment (ManeraKai) [reply org.libredirect.frontends_manager.yml:18]: So in other words I don't need the permission `--filesystem=xdg-data` to access `$HOME/.var/app/<app-id>/data`?

File: org.libredirect.frontends_manager.metainfo.xml
Comment (ManeraKai) [reply org.libredirect.frontends_manager.metainfo.xml:1]: Fixed.

File: org.libredirect.frontends_manager.yml (hunk 7 above)
Comment (ManeraKai) [reply org.libredirect.frontends_manager.yml]: Fixed.

File: org.libredirect.frontends_manager.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.libredirect.frontends_manager.yml]: no.

File: CONTRIBUTING.md
Comment (hfiguiere) [inline CONTRIBUTING.md:1]: you should also remove the `.github` because you submitted the PR against the wrong branch. (it supposed to be agains `new-pr`)

File: org.libredirect.frontends_manager.metainfo.xml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      VPN.
+    </p>
+  </description>
+  <content_rating type="oars-1.1" />
+
+  <launchable type="desktop-id">libredirect-frontends-manager.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>
+        https://raw.githubusercontent.com/libredirect/frontends_manager/main/src/assets/imgs/screenshot_1.png</image>
```
Comment (hfiguiere) [inline org.libredirect.frontends_manager.metainfo.xml:28]: so is the fail disappear on `main` this will break. You should use a link to the tag or commit.

File: screenshots/Main Window.png
Comment (hfiguiere) [inline screenshots/Main Window.png:1]: why is that file here? It's never used.

File: org.libredirect.frontends_manager.yml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+      env:
+        npm_config_nodedir: /usr/lib/sdk/node16
+        CARGO_HOME: /run/build/libredirect-frontends-manager/cargo
+        npm_config_offline: 'true'
+        npm_config_cache: /run/build/libredirect-frontends-manager/flatpak-node/npm-cache
+    build-commands:
+      - npm ci --offline
+      - cargo --offline fetch --manifest-path src-tauri/Cargo.toml
+      - npm run tauri build -- -d -b deb
+      - install -Dm644 -t /app/share/metainfo/ org.libredirect.frontends_manager.metainfo.xml
```
[the remark this answers is not in the document]
Comment (ManeraKai) [reply org.libredirect.frontends_manager.yml:47]: Fixed.

**Comment by ManeraKai** [comment]
Should I have a hyphen or an underscore in thea Apllication id?

**Comment by bbhtt** [comment]
What's happening with the id? libredirect.org doesn't resolve to anything

> https://docs.flathub.org/docs/for-app-authors/requirements#application-id

The Application ID should be a real URL of a domain that the app author has control over or where their app is hosted.

**Comment by ManeraKai** [comment]
Alright updated the Application id to `io.github.libredirect.frontends-manager`

File: io.github.libredirect.frontends-manager.yml [hunk 12]
```diff
@@ -0,0 +1,51 @@
+id: io.github.libredirect.frontends-manager
+runtime: org.gnome.Platform
+runtime-version: '42'
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.yml:3]: you should build webkist gtk from source to get the right (unsupported) version. Also you should lobby tauri to fix this. They would rush a fix if it was the proprietary platform. For now they are just building something insecure since it requires to use no-longer supported webengine.

**Comment by ManeraKai** [comment]
So?

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by bbhtt** [comment]
You need to make a PR with explanation for an exception, for the build to pass https://github.com/flathub/flatpak-builder-lint/pulls

**Comment by ManeraKai** [comment]
@bbhtt https://github.com/flathub/flatpak-builder-lint/pull/170

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by bbhtt** [comment]
It won't work until that is merged.

**Comment by bbhtt** [comment]
Did you build the flatpak and run it locally? I believe you would also need to access the host docker socket `/run/docker.sock`, `/run/docker` or `xdg-run/docker.{sock}` - something like that, but I'm not sure.

**Comment by ManeraKai** [comment]
Yes I did, but with Docker Desktop I didn't need docker.socket, or let me say it wasn't there in thle first place. I'mexecuting commands through the docker cli directly.

**Comment by ManeraKai** [comment]
Well yesterday I found out my docker socket was actually in `~/.docker/desktop/docker.socket` when using Docker Desktop. So I'll see what I can do to remove `--talk-name=org.freedesktop.Flatpak`

**Comment by ManeraKai** [comment]
It needs a code rewrite. I'll close the PR till I finish it.

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by hfiguiere** [comment]
package submission PR are supposed to be done against the `new-pr` branch

**Comment by ManeraKai** [comment]
So I'll make a new PR.

File: io.github.libredirect.frontends-manager.metainfo.xml [hunk 13]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.libredirect.frontends_manager</id>
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.metainfo.xml:3]: this doesn't match

File: io.github.libredirect.frontends-manager.yml (hunk 12 above)
Comment (hfiguiere) [reply io.github.libredirect.frontends-manager.yml:3]: This runtime is EOL. Use 45.

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

File: io.github.libredirect.frontends-manager.yml [hunk 14]
```diff
@@ -0,0 +1,84 @@
+id: io.github.libredirect.frontends-manager
+runtime: org.gnome.Platform
+runtime-version: '44'
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.yml:3]: 45 is the current runtime

**Comment by ManeraKai** [comment]
> the build still fail on aarch64.

I'm working on that.

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

File: io.github.libredirect.frontends-manager.yml [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    build-options:
+      append-path: /usr/lib/sdk/node16/bin:/usr/lib/sdk/rust-stable/bin
+      env:
+        npm_config_nodedir: /usr/lib/sdk/node16
+        CARGO_HOME: /run/build/libredirect-frontends-manager/cargo
+        npm_config_offline: 'true'
+        npm_config_cache: /run/build/libredirect-frontends-manager/flatpak-node/npm-cache
+    build-commands:
+      - npm ci --offline
+      - ls /usr/lib/sdk
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.yml:76]:
```suggestion
```

**Comment by ManeraKai** [comment]
Can I install the module [@tauri-apps/cli-linux-arm64-gnu](https://www.npmjs.com/package/@tauri-apps/cli-linux-arm64-gnu) when I'm building on aarch64?

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
Queued test build for io.github.libredirect.frontends-manager.

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

File: io.github.libredirect.frontends-manager.yml [hunk 16]
```diff
@@ -0,0 +1,84 @@
+id: io.github.libredirect.frontends-manager
+runtime: org.gnome.Platform
+runtime-version: '44'
+command: libredirect-frontends-manager
+rename-icon: libredirect-frontends-manager
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.yml:5]: except that you install the icon as `icon`....

File: io.github.libredirect.frontends-manager.yml [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+        npm_config_offline: 'true'
+        npm_config_cache: /run/build/libredirect-frontends-manager/flatpak-node/npm-cache
+
+    build-commands:
+      - npm ci --offline
+      - cargo --offline fetch --manifest-path src-tauri/Cargo.toml
+      - npm run tauri build -- -b none
+      - install -Dm644 -t /app/share/metainfo/ io.github.libredirect.frontends-manager.metainfo.xml
+      - install -Dm755 -t /app/bin/ src-tauri/target/release/libredirect-frontends-manager
+      - install -Dm755 -t /app/share/icons/hicolor/scalable/apps/ src-tauri/icons/icon.svg
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.yml:83]:
```suggestion
      - install -Dm644 src-tauri/icons/icon.svg /app/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg
```

and remove the `rename-icon`

File: io.github.libredirect.frontends-manager.desktop [hunk 18]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Categories=Utility;
+Comment=Run privacy frontends on your computer
+Exec=libredirect-frontends-manager
+Icon=libredirect-frontends-manager
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.desktop:5]:
```suggestion
Icon=io.github.libredirect.frontends-manager
```

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

File: io.github.libredirect.frontends-manager.metainfo.xml [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+      Normally you use LibRedirect browser extension to redirect from e.g reddit.com/post =>
+      libreddit-instance.xyz/post. This does have problems like with reddit limiting requests from a
+      single ip, to you sending your data to an unkown instance. LibRedirect Frontends Manager let's
+      you to run those frontends locally on your computer. For anonymizing your IP, use Lokinet or a
+      VPN.
+    </p>
+  </description>
+  <content_rating type="oars-1.1" />
+
+  <launchable type="desktop-id">libredirect-frontends-manager.desktop</launchable>
```
Comment (hfiguiere) [inline io.github.libredirect.frontends-manager.metainfo.xml:28]:
```suggestion
  <launchable type="desktop-id">io.github.libredirect.frontends-manager.desktop</launchable>
```

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by bbhtt** [comment]
Update the runtime to 45 please.

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
bot, build io.github.libredirect.frontends-manager

**Comment by ManeraKai** [comment]
@bbhtt Done.

**Comment by ManeraKai** [comment]
Is there anything else I need to do for it to be merged?

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
bot, build io.github.libredirect.frontends-manager
